src/Arpack.jl

```diff
@@ -310,6 +310,11 @@ julia> s.S
     that the size is smallest.
 """
 svds(A; kwargs...) = _svds(A; kwargs...)
+function _orth!(P)
+    Q,R = qr!(P)
+    rsign = [iszero(rval) ? one(rval) : sign(rval) for rval in diag(R)]
```

Comment:
Minor optimization but you can avoid the allocation from `diag` with something like
```julia
_sign(x) = iszero(x) ? one(x) : sign(x)
rsign = [_sign(R[i,i]) for i in 1:size(R, 2)]
```

Reply:
Ah right, that's clean. Added!